packages/waku/src/lib/middleware/handler.ts

```diff
@@ -35,9 +35,10 @@ const getInput = async (
   }
   const basePrefix = config.basePath + config.rscBase + '/';
   if (ctx.req.url.pathname.startsWith(basePrefix)) {
-    const rscPath = decodeRscPath(
+    let rscPath = decodeRscPath(
       decodeURI(ctx.req.url.pathname.slice(basePrefix.length)),
     );
+    rscPath = rscPath.split('/').map(encodeURIComponent).join('/');
```

Comment:
@dai-shi let me know how you'd like this to be cleaned up or fixed differently, but this makes it so the path passed through our router as a page prop is consistently encoded.